crypto/context.c

```diff
@@ -80,62 +93,261 @@ static int context_init(OSSL_LIB_CTX *ctx)
     if (ctx->oncelock == NULL)
         goto err;
 
-    for (i = 0; i < OSSL_LIB_CTX_MAX_INDEXES; i++) {
-        ctx->index_locks[i] = CRYPTO_THREAD_lock_new();
-        ctx->dyn_indexes[i] = -1;
-        if (ctx->index_locks[i] == NULL)
-            goto err;
-    }
+    ctx->rand_crngt_lock = CRYPTO_THREAD_lock_new();
+    if (ctx->rand_crngt_lock == NULL)
+        goto err;
 
-    /* OSSL_LIB_CTX is built on top of ex_data so we initialise that directly */
+    /* Initialize ex_data. */
     if (!ossl_do_ex_data_init(ctx))
         goto err;
     exdata_done = 1;
 
-    if (!ossl_crypto_new_ex_data_ex(ctx, CRYPTO_EX_INDEX_OSSL_LIB_CTX, NULL,
-                                    &ctx->data))
+    /* P2. We want evp_method_store to be cleaned up before the provider store */
```

Comment:
I'd prefer P2 to be _Highest priority_.
Likewise P1 to be _High priority_.

It took me too long to figure out what was meant.

Below uses _default_ and _low_.  I guess P0 and P-1 would work for them but they are less obvious IMO.